packages/logger/src/log/LogItem.ts

```diff
@@ -31,7 +30,14 @@ class LogItem implements LogItemInterface {
   }
 
   public removeEmptyKeys(attributes: LogAttributes): LogAttributes {
-    return pickBy(attributes, (value) => value !== undefined && value !== '' && value !== null);
+    const newAttributes: LogAttributes = {};
+    for (const key in attributes) {
+      if (attributes[key] !== undefined && attributes[key] !== '' && attributes[key] !== null) {
+        newAttributes[key] = attributes[key];
+      }
+    }
+    
+    return newAttributes;
   }
```

Comment:
Firstly, I wrote a declarative one-liner:
```
public removeEmptyKeys(attributes: LogAttributes): LogAttributes {
    return Object.fromEntries(Object.entries(attributes).filter(([, value]) => value !== undefined && value !== '' && value !== null));
}
```
But for the sake of readability, I stick to the verbose version.

Reply:
I don't know if nested objects could be in the `logItem`, but it wasn't and isn't supported by this function.

Reply:
Could you give an example of what do you mean by nested objects?

Reply:
`logger.info("my message", { value: 42, nested: { value: "nested value", emptyValue: '', undefinedValue: undefined, nullValue: null } });`

Reply:
I think we need to support nested objects though. 

Here's an example of logs emitted by Logger `v1.5.0` (latest):
![Screenshot 2022-12-28 at 14 40 48](https://user-images.githubusercontent.com/7353869/209821310-cdd91bd6-6b5b-49b5-a128-b5df6e7203d0.png)
 
from this code:
```ts
import { Logger, injectLambdaContext } from "@aws-lambda-powertools/logger";
import middy from "@middy/core";

const logger = new Logger({});

export const handler = middy(async () => {
  logger.info("my message", {
    value: 42,
    nested: {
      value: "nested value",
      emptyValue: "",
      undefinedValue: undefined,
      nullValue: null,
    },
  });
}).use(injectLambdaContext(logger, { logEvent: true }));
```

this was the event payload passed:
```json
{
  "key1": "value1",
  "key2": "value2",
  "key3": "value3",
  "nested": {
      "some": "key",
      "other": 1
  }
}
```

We need to support nested objects both for the `logEvent` functionality (also present in the unit tests) and arbitrary objects

Reply:
Oh after re-reading I realize I might have misunderstood your comment.

Were you referring to supporting the removal of `undefined`, `null`, etc. from nested objects?

Reply:
> Oh after re-reading I realize I might have misunderstood your comment.
> 
> Were you referring to supporting the removal of `undefined`, `null`, etc. from nested objects?

yes, it's just go through the first level

Reply:
I see, apologies for the misunderstanding!

I think that at the moment we should strive for maintaining the same exact functionality level as the current version without breaking changes. Which based on the test I did above, seems to at least remove `undefined` values.

Then later on, if there's demand we can explore extending the behavior to empty strings and/or `null`.

Reply:
`undefined` values are filtered somewhere else 🙃

Reply:
And I definitely support/agree with the choice of using a more verbose version for the sake of readability.